mars/dataframe/datasource/read_raydataset.py

```diff
@@ -0,0 +1,115 @@
+    _op_type_ = OperandDef.READ_RAYDATASET
+
+    _refs = AnyField('refs')
+    _columns = ListField('columns')
+    _incremental_index = BoolField('incremental_index')
+    _nrows = Int64Field('nrows')
+
+    def __init__(self, refs=None, columns=None,
+                 incremental_index=None, nrows=None,
+                 **kw):
+        super().__init__(_refs=refs, _columns=columns,
+                         _incremental_index=incremental_index,
+                         _nrows=nrows,
+                         **kw)
+
+    @property
+    def refs(self):
+        return self._refs
+
+    @property
+    def columns(self):
+        return self._columns
+
+    @property
+    def incremental_index(self):
+        return self._incremental_index
+
+    @classmethod
+    def _tile_partitioned(cls, op: "DataFrameReadRayDataset"):
+        out_df = op.outputs[0]
+        shape = (np.nan, out_df.shape[1])
+        dtypes = out_df.dtypes
+        dataset = op.refs
+
+        chunk_index = 0
+        out_chunks = []
+        for piece in dataset:
+            chunk_op = op.copy().reset_key()
+            chunk_op._refs = piece
+            new_chunk = chunk_op.new_chunk(
+                None, shape=shape, index=(chunk_index, 0),
+                index_value=out_df.index_value,
+                columns_value=out_df.columns_value,
+                dtypes=dtypes)
+            out_chunks.append(new_chunk)
+            chunk_index += 1
+
+        new_op = op.copy()
+        nsplits = ((np.nan,) * len(out_chunks), (out_df.shape[1],))
+        return new_op.new_dataframes(None, out_df.shape, dtypes=dtypes,
+                                     index_value=out_df.index_value,
+                                     columns_value=out_df.columns_value,
+                                     chunks=out_chunks, nsplits=nsplits)
+
+    @classmethod
+    def _tile(cls, op):
+        return cls._tile_partitioned(op)
+
+    @classmethod
+    def execute(cls, ctx, op: "DataFrameReadRayDataset"):
+        out = op.outputs[0]
+        refs = op.refs
+
+        df = ray.get(refs)
+        ctx[out.key] = df
+
+    def __call__(self, index_value=None, columns_value=None, dtypes=None):
+        shape = (np.nan, len(dtypes))
+        return self.new_dataframe(None, shape, dtypes=dtypes, index_value=index_value,
+                                  columns_value=columns_value)
+
+
+def read_raydataset(refs, columns=None,
+                    incremental_index=False,
+                    **kwargs):
+    dtypes = ray.get(ray.remote(_get_dtypes).remote(refs[0]))
+    index_value = parse_index(pd.RangeIndex(-1))
+    columns_value = parse_index(dtypes.index, store_data=True)
+
+    op = DataFrameReadRayDataset(refs=refs, columns=columns,
+                                 incremental_index=incremental_index)
+    return op(index_value=index_value, columns_value=columns_value,
+              dtypes=dtypes)
+
+
+def _get_dtypes(t):
```

Comment:
For ray dataset, the ray object is an arrow.Table, does it has `dtypes`? DO we neec to read arrow table schema? Or just use the schema in ray.data.datset.DataSet